src/config/test/config_tests.erl

```diff
@@ -115,6 +111,41 @@ handle_config_terminate(Self, Reason, {Pid, State}) ->
     Pid ! {config_msg, {Self, Reason, State}},
     ok.
 
+setup_ini(IniFiles) ->
+    Chain = [write_ini(F) || F <- IniFiles],
+    {Chain, setup(Chain)}.
+
+teardown_ini(_, {Chain, Apps}) ->
+    lists:foreach(
+        fun(Path) -> ok = file:delete(Path) end,
+        Chain
+    ),
+    teardown(Apps).
+
+write_ini(FileName) ->
+    Path = filename:join([?TEMPDIR, FileName]),
+    Data = io_lib:format("[section]\nfile = ~s\n", [FileName]),
+    ok = file:write_file(Path, Data),
+    Path.
+
+config_delete_reload_restart_test_() ->
+    {
+        "Test consistency after set, delete, reload, and restart",
+        foreachx,
+        fun setup_ini/1,
+        fun teardown_ini/2,
+        [
+            {
+                ["default.ini", "local.ini"],
+                fun non_persistent_set_delete_reload_restart/2
```

Comment:
I tried adding these lines to `couch_eunit.hrl`:
```erlang
-ifndef(TDEF_FEX).
-define(TDEF_FEX(Name), fun(R, X) -> {atom_to_list(Name), ?_test(Name(R, X))} end).
-define(TDEF_FEX(Name, Timeout), fun(R, X) -> {atom_to_list(Name), {timeout, Timeout, ?_test(Name(R, X))}} end).
-endif.
```
and rewrote these new tests to use it.

However, as I worked on converting the _rest_ of this file to use `TDEF_FEX` in the `foreachx` tests, I realized it might take a bit more effort with tests like e.g. `should_notify_on_config_reload` which don't seem to lend themselves to a simple conversion. I will continue to ponder the best approach, and would welcome advice.

Reply:
For one or two extra parameters, the complexity of a foreachx might not be worth it? For the two options just having two test setups: `setup_local_ini` and `setup_default_ini` and they can call the same setup `setup_ini/1` utility function could be simpler and could use the standard `?TDEF_FE` macro?

Reply:
Those tests originally did just use `foreach`, and the generated `.ini` files were hardcoded in `setup_ini`. But then I was curious about whether persistent `set` and `delete` variations would also write to an `.ini` file if there was only one, so I changed it to `foreachx` in order to factor out the config chain. Turns out that because of how `write_filename` is defined:
```erlang
    WriteFile =
        case IniFiles of
            [_ | _] -> lists:last(IniFiles);
            _ -> undefined
        end,
```
that it does indeed write to a single `.ini` file configuration (further evidence that `default.ini` is not a great name).

So maybe I should just change it back now that my experiment has concluded there's no real edge case for just a single `.ini` config?

Reply:
> My main beef with `foreachx` is that it becomes another eunit quirk one has to learn to be able add add or fix failing tests.

I find it annoying that it requires `Pairs::[{X::any(), ((X::any(), R::any()) -> Tests)}]` instead of (say) an `Instantiator` that returns `Pairs`. But, yeah, I agree we can keep its use to a minimum. I've changed those tests back to `foreach`.